src/gt4py/stencil_object.py

```diff
@@ -322,15 +322,17 @@ def _call_run(
             origin = normalize_origin_mapping(origin)
 
         for name, field in used_field_args.items():
+            storage_ndim = len(field.shape)
             if "_all_" in origin:
                 field_mask = self._get_field_mask(name)
-                origin.setdefault(name, gt_ir.Index(origin["_all_"].filter_mask(field_mask)))
+                field_origin = gt_ir.Index(origin["_all_"].filter_mask(field_mask))
+                min_origin = [min(field.shape[i] - 1, field_origin[i]) for i in range(storage_ndim)]
```

Comment:
My gut instinct tells me something is wrong here. We're taking the min of a shape and origin, so the "units" seem off. Is this intended? If so, perhaps a comment explaining why would help.

Reply:
I also don't fully get it. My guess is that the intention here is to catch errors were the provided field shape is even smaller that the `origin` point, right? If that's the case and there is not another check in place for this possibility, I think it's more reasonable to leave the original code as it was and add a check raising an exception if it fails.

Reply:
Thank you Enrique, I have moved the fix for this into our dynamical core code since that is where the problem arises. This PR is only error message updates now.